frontend/src/metabase/api/user.ts

```diff
@@ -0,0 +1,68 @@
+import type {
+  CreateUserResponse,
+  CreateUserRequest,
+  ListUsersRequest,
+  ListUsersResponse,
+  UpdatePasswordRequest,
+  UserId,
+} from "metabase-types/api";
+
+import { Api } from "./api";
+// import { listTag, invalidateTags } from "./tags";
+
+export const userApi = Api.injectEndpoints({
+  endpoints: builder => ({
+    listUsers: builder.query<ListUsersResponse, ListUsersRequest | null>({
+      query: body => ({
+        method: "GET",
+        url: "/api/user",
+        body,
+      }),
+      // FIXME: providesTags is needed
+    }),
+    listUserRecipients: builder.query({
+      query: () => ({
+        method: "GET",
+        url: "/api/user/recipients",
+      }),
+      // FIXME: is providesTags needed ??
+    }),
+    createUser: builder.mutation<CreateUserResponse, CreateUserRequest>({
+      query: body => ({
+        method: "POST",
+        url: "/api/user",
+        body,
+      }),
+      // FIXME: invalidatesTags is needed
+    }),
+    updatePassword: builder.mutation<void, UpdatePasswordRequest>({
+      query: ({ id, old_password, password }) => ({
+        method: "PUT",
+        url: `/api/user/${id}/password`,
+        body: { old_password, password },
+      }),
+      // FIXME: invalidatesTags is maybe needed?
+    }),
+    deactivateUser: builder.query<void, { id: UserId }>({
```

Comment:
Other new endpoints would just accept `UserId`, not `{id: UserId}`.
Let's make this consistent?
Same goes for `reactivateUser`.

Reply:
https://github.com/metabase/metabase/pull/41312/commits/e9498124a1bb676d2d563f51d59168796e597a56 and then https://github.com/metabase/metabase/pull/41312/commits/6369c71bf71923a22765ed14f54f3ae4592d6ecc